apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/SessionPoolTests.TargetedSessionPoolTests.cs

```diff
@@ -928,6 +929,21 @@ public async Task ShutdownPoolAsync_CannotAcquireAfterShutdown()
                 });
             }
 
+            [Fact]
+            public async Task BatchCreateSession_WithDatabaseRoleAsync()
+            {
+                var spannerClientMock = new SessionTestingSpannerClient();
+                var parent = new SessionPool(spannerClientMock, new SessionPoolOptions());
+                var targetedSessionPool = new TargetedSessionPool(parent, SessionPoolKey.Create(s_databaseName).WithDatabaseRole(s_databaseRole), acquireSessionsImmediately: true);
+
+                // Give the pool a minute to fill up
+                await spannerClientMock.Scheduler.RunForSecondsAsync(60);
+
+                BatchCreateSessionsRequest batchCreateSessionsRequest;
+                Assert.True(spannerClientMock.BatchCreateSessionRequests.TryDequeue(out batchCreateSessionsRequest));
+                Assert.Equal(s_databaseRole, batchCreateSessionsRequest.SessionTemplate.CreatorRole);

```

Comment:
Assert the database name as well, just to make sure that `SessionPoolKey.WithDatabaseRole` doesn't lose that?